connectivity/check/check.go

```diff
@@ -150,6 +150,35 @@ type FlowParameters struct {
 
 type flowsSet []*observer.GetFlowsResponse
 
+// lastTime returns the timestamp of the last flow in 'f', if any.
+func (f flowsSet) lastTime() (last time.Time) {
+	if len(f) == 0 {
+		return last
+	}
+	timestamp := f[len(f)-1].GetFlow().GetTime()
+	if timestamp == nil {
+		return last
+	}
+	return timestamp.AsTime()
+}
+
+// append adds any new flows in 'from' to 'f' and returns the combined set
+func (f flowsSet) append(from flowsSet) flowsSet {
+	last := f.lastTime()
+	if last.IsZero() {
+		return from
+	}
+	// Add all new flows with a later timestamp
+	// XXX: Hubble should not return flows with the same timestamp in a later flow response as before
```

Comment:
What does `XXX` mean here?

Reply:
Addressed this by expanding the function comment as follows:
```
+// append adds any flows in 'from' to 'f', if the flow's timestamp is
+// later than the last one in 'f', returns the combined set.
```
This logic would fail to add a flow if 'from' contains flows with the same timestamp as the last in 'f'. My understanding is that when 'from' is a fresh batch of flows received from Hubble this should not happen.